iree/turbine/kernel/wave/templates/paged_decode_attention.py

```diff
@@ -292,30 +295,40 @@ def repeat(
             res, output, mapping=mapping, elements_per_thread=PHASE_1_ELEMS_PER_THREAD
         )
 
+    (
+        num_query_heads,
+        num_kv_heads,
+        head_size,
+        head_size_kv,
+        block_size,
+        num_seqs,
+        kv_lens,
+    ) = shape
+
     symbols_0 = {
         ADDRESS_SPACE: SHARED_ADDRESS_SPACE,
         LOAD_ELEMS_PER_THREAD: get_mfma_load_elems_per_thread(mfma_variant),
         STORE_ELEMS_PER_THREAD: get_mfma_store_elems_per_thread(mfma_variant),
         BLOCK_BH: 1,
-        BLOCK_B: shape[0] // shape[5],
+        BLOCK_B: num_query_heads // num_kv_heads,
         BLOCK_S: 1,
         BLOCK_U: 1,
-        B: shape[0],
-        M: shape[1],
-        N: shape[2],
-        K1: shape[3],
-        K2: shape[4],
-        BH: shape[5],
-        S: shape[6],
+        B: num_query_heads,
+        M: 1,
+        N: head_size_kv,
+        K1: head_size,
+        K2: block_size,
+        BH: num_kv_heads,
+        S: num_seqs,
```

Comment:
Commenting separately because GitHub doesn't seem to allow me to mark the comment unresolved:

Now that shape is a NamedTuple you could also access these all by name rather than unpacking shape first.